docs/oauth2.md

```diff
@@ -538,13 +664,94 @@ In the following RAR example
+auth_oauth2.https.cacertfile = /opts/certs/prodcacert.pem
+```
+
+And the equivalent advanced configuration:
+
+```ini
+auth_oauth2.resource_server_id = rabbitmq-prod
+auth_oauth2.scope_prefix = rabbitmq.
+auth_oauth2.default_oauth_provider = prodkeycloak
+
+auth_oauth2.oauth_providers.prodkeycloak.issuer = https://prodkeycloak:8080/realm/prod
+auth_oauth2.oauth_providers.prodkeycloak.https.cacertfile = /opts/certs/prodcacert.pem
+```
+
+This advanced configuration will be more relevant when users present tokens which have been issued or signed by different OAuth 2.0 providers. However, one can still use it provided `auth_oauth2.default_oauth_provider` is set.
+
+### Multiple Resource Servers {#multiple-resource-servers}
+
+Typically, all users that access a RabbitMQ cluster are registered within the same identity provider. And likewise, all tokens targeting the same RabbitMQ cluster also carry the same *audience*. In other words, all users reference a RabbitMQ cluster with the same resource name which must match the value of the `auth_oauth2.resource_server_id` setting.
+
+However, there are some use-cases where RabbitMQ is accessed by users registered in different Identity Providers or tokens are issued for the same RabbitMQ installation but with different *Audience*(s). For these use-cases, RabbitMQ OAuth 2.0 plugin and the Management plugin can be configured with multiple OAuth 2.0 resources.
+
+Below is the OAuth 2.0 plugin configuration for two resources with the ids, `rabbit_prod` and `rabbit_dev`. Both resources (a.k.a. *audience*) are managed by the same Identity Provider whose issuer is `http//some_idp_url`.
+
+```ini
+auth_oauth2.issuer = http//some_idp_url
+auth_oauth2.scope_prefix = rabbitmq.
+
+auth_oauth2.resource_servers.1.id = rabbit_prod
+auth_oauth2.resource_servers.2.id = rabbit_dev
+auth_oauth2.resource_servers.2.scope_prefix = dev-rabbitmq.
+```
+
+All resource servers share the settings you have set so far under `auth_oauth2.` such as `scope_prefix`. However, they can override them. Here are the list of settings a resource server can override:
+- `id` - This is the actual resource identifier carried in the `audience` field of a token. If omitted, the value is the index, e.g. given `auth_oauth2.resource_servers.prod.scope_prefix` setting, the `id` would be `prod`.
+- `scope_prefix`
+- `additional_scopes_key`
+- `resource_server_type`
+- `oauth_provider_id` - This is the identifier of the OAuth provider, configured in RabbitMQ, which provides all the settings to contact the authorization server in order to discover all its endpoints, such as the `jwks_uri` to download the signing keys to validate the token. If this setting is omitted, RabbitMQ looks up the default Authorization Provider's id in the setting `auth_oauth2.default_oauth_provider`. And if it is also omitted, RabbitMQ uses `auth_oauth2.issuer` or `auth_oauth2.jwks_url` to download the signings keys to validate the token.
```

Comment:
ok the first sentence is too long, you can break it up like this "This is the identifier of the OAuth provider. It is configured in RabbitMQ. It provides.....

no need for the word "in order" 

again I wouldn't start a sentence with "And" just run the sentence above on. "........id in the setting `auth_oauth2.default_oauth_provider`, and if it is also omitted..."